plate_mapper/tests/test_plate_mapper.py

```diff
@@ -9,200 +9,75 @@
 
 
 class PlateMapperTests(TestCase):
     """ Tests for plate_mapper.py """
 
     def setUp(self):
-        """ Set up working directory and test files
-        """
+        """ Create working directory """
         self.wkdir = mkdtemp()
-        self.barseq_temp_fp = join(self.wkdir, "barseq_temp.tsv")
-        with open(self.barseq_temp_fp, 'w') as tmp:
-            tmp.write(barseq_temp)
-        self.plate_map_fp = join(self.wkdir, "plate_map.tsv")
-        with open(self.plate_map_fp, 'w') as tmp:
-            tmp.write(plate_map)
-        self.mapping_fp = join(self.wkdir, "mapping.tsv")
-        self.plate_map_cherr_fp = join(self.wkdir, "plate_map_cherr.tsv")
-        with open(self.plate_map_cherr_fp, 'w') as tmp:
-            tmp.write(plate_map_col_header_err)
-        self.plate_map_rherr_fp = join(self.wkdir, "plate_map_rherr.tsv")
-        with open(self.plate_map_rherr_fp, 'w') as tmp:
-            tmp.write(plate_map_row_header_err)
-        # list of files to remove
-        self.tmpfiles = [self.barseq_temp_fp,
-                         self.plate_map_fp,
-                         self.mapping_fp,
-                         self.plate_map_cherr_fp,
-                         self.plate_map_rherr_fp]
 
     def tearDown(self):
-        for file in self.tmpfiles:
-            if isfile(file):
-                remove(file)
+        """ Delete working directory """
         rmtree(self.wkdir)
 
-    def test__main(self):
+    def test_plate_mapper(self):
         # test a successful conversion
-        input_f = open(self.plate_map_fp, 'r')
-        barseq_f = open(self.barseq_temp_fp, 'r')
-        output_f = open(self.mapping_fp, 'w')
-        plate_mapper(input_f, barseq_f, output_f)
-        with open(self.mapping_fp, 'r') as input_f:
-            obs = input_f.read()
-        self.assertEqual(obs, exp_mapping)
-        # test column header Error
-        input_f = open(self.plate_map_cherr_fp, 'r')
+        # subdirectory "data", in which test data files are located
+        datadir = join(dirname(realpath(__file__)), 'data')
+        # a simplified plate map file containing two plates, each having
+        # 4 columns x 3 rows + 2 property fields
+        input_f = open(join(datadir, 'plate_map.txt'), 'r')
+        # a simplified barcode sequence template file containing 4 plates
+        barseq_f = open(join(datadir, 'barseq_temp.txt'), 'r')
+        # the mapping file to be written
+        obs_output_fp = join(self.wkdir, 'obs_mapping.txt')
+        output_f = open(obs_output_fp, 'w')
+        # expected output mapping file
+        exp_output_fp = join(datadir, 'exp_mapping.txt')
+        # sample name list
+        names_f = open(join(datadir, 'sample_list.txt'), 'r')
+        with self.assertRaises(SystemExit) as context:
+            plate_mapper(input_f, barseq_f, output_f, names_f)
+        # check output mapping file
+        with open(obs_output_fp, 'r') as f:
+            obs = f.read()
```

Comment:
It looks like this file didn't get deleted after construction.  I'd take a look at python setUp and tearDown